src/postings/postings_writer.rs

```diff
@@ -46,37 +47,38 @@ fn make_field_partition(
 /// It pushes all term, one field at a time, towards the
 /// postings serializer.
 pub(crate) fn serialize_postings(
-    ctx: IndexingContext,
+    ctx: &IndexingContext,
     schema: Schema,
     per_field_postings_writers: &PerFieldPostingsWriter,
     fieldnorm_readers: FieldNormReaders,
     doc_id_map: Option<&DocIdMapping>,
+    unordered_id_to_ordered_id: &[(OrderedPathId, TinySet)],
     serializer: &mut InvertedIndexSerializer,
 ) -> crate::Result<()> {
     // Replace unordered ids by ordered ids to be able to sort
-    let unordered_id_to_ordered_id: Vec<OrderedPathId> =
-        ctx.path_to_unordered_id.unordered_id_to_ordered_id();
+    let ordered_id_to_path = ctx.path_to_unordered_id.ordered_id_to_path();
```

Comment:
Why move the binding up here if it is not used in the loop? I would suggest leaving it below to make that clear to the reader.

Reply:
The Vec is reused in another method

Reply:
Isn't it `unordered_id_to_ordered_id` that is reused in `serialize_segment_fields`? I was thinking about the binding for `ordered_id_to_path` which moved up here from line 79 without any change to how it is used.